### ISSUE TITLE
Question of the $L_{E}$ in eq 10

### ISSUE THREAD

**Issue by OverfitFlow** [tsosea2/MarginMatch#2]
Hi,

I have read your codes carefully, and have some questions about  
line 141 of models/marginmatch/marginmatch_utils.py
`both_mask = both_mask * inverse_mask + threshold_mask.float() * (sum(saved_both_mask) / logits_w.shape[0])`

I assume this line is the denominator to do the averaging for the loss.
so regarding the part for equation 10, `sum(saved_both_mask) / logits_w.shape[0]` should be
`logits_w.shape[0] / sum(saved_both_mask)` instead.

since the value of 'logits_w.shape[0]' will be offset when return the mean value, 
and I am still confused about the value 'sum(saved_both_mask)', is this should be replaced by another value?

Please correct me if I am wrong, thank you.

**Comment by tsosea2** [comment]
Hello,

Thanks for the question! That term ensures that during the early stages of training when few examples pass the threshold the loss of the threshold examples is scaled down based on how many examples pass the threshold. `(sum(saved_both_mask) / logits_w.shape[0])` simply scales down the threshold loss based on this (how_many_examples_pass_the_threshold / batch_size).
